### PR TITLE
fix: improve stalled retrieval cancellation

### PR DESCRIPTION
**Pull request by LexLuthr**

### CODE REVIEW

File: retrievalmarket/rtvllog/retrieval_log.go [hunk 1]
```diff
@@ -12,6 +14,8 @@ import (
 	logging "github.com/ipfs/go-log/v2"
 )
 
+var ErrRetrievalNotFound = fmt.Errorf("No channel")
```
[the remark this answers is not in the document]
Comment (LexLuthr) [reply retrievalmarket/rtvllog/retrieval_log.go:17]: I missed the file name. Fixing.

File: retrievalmarket/server/gsunpaidretrieval.go [hunk 2]
```diff
@@ -175,7 +176,7 @@ func (g *GraphsyncUnpaidRetrieval) CancelTransfer(ctx context.Context, id datatr
 
 	if state == nil {
 		g.activeRetrievalsLk.Unlock()
-		return fmt.Errorf("no transfer with id %d", id)
+		return fmt.Errorf("%w with id: %d", ErrRetrievalNotFound, id)
```
Comment (dirkmc) [inline retrievalmarket/server/gsunpaidretrieval.go:179]: I think the error needs to be at the end in order for it to be wrapped properly:
return fmt.Errorf("transfer id %d: %w", id, ErrRetrievalNotFound)

**Comment by dirkmc** [comment]
Sorry I realized that we need to set a short timeout for cancelling the unpaid retrieval only: looking up the retrieval is fast, but sending the cancel message may be slow.

If we set a short timeout for cancelling legacy retrievals, it takes so long to look up the retrieval that the context may time out before we get to cancel the retrieval. So those retrievals might never get cleaned up.

I pushed a commit to only set the short timeout for cancelling unpaid retrievals.

**Comment by dirkmc** [comment]
I think we should test it first - I want to make a change to enable serving indexing announcements over http then we can test them together

**Pull request merged**